packages/core/src/pipelines/event.pipeline.ts

```diff
@@ -0,0 +1,152 @@
+            throw new EventMappingError("There are no events to execute.", event, interceptedEvent, executionContext)
+        }
+
+        const eventsExecutionPromises: Promise<EventResponse<any, any> | EventResponse<any, any>[]>[] = [];
+
+        // 3- Loop over the EventExecutionOptions array and start executing the events
+        eventExecutions.forEach(eventExecutionOptions => {
+            switch (eventExecutionOptions.executionOrder) {
+                case 'sequential':
+                    eventsExecutionPromises.push(new Promise<EventResponse<any, any>[]>(async resolve => {
+                        // await all events and then resolve.
+                        const eventResponses = [];
+
+                        for (const event of eventExecutionOptions.events) {
+                            const childContainer = container.createChildContainer();
+                            const eventDispatcher = childContainer.resolve("EventDispatcherInterface") as EventDispatcherInterface;
+
+                            eventResponses.push(await this.executeEvent(event, eventDispatcher));
+                        }
+
+                        return resolve(eventResponses);
+                    }))
+                    break;
+                case 'parallel':
+                    for (const event of eventExecutionOptions.events) {
+                        eventsExecutionPromises.push(new Promise<EventResponse<any, any> | EventResponse<any, any>[]>(async resolve => {
+                            const childContainer = container.createChildContainer();
+                            const eventDispatcher = childContainer.resolve("EventDispatcherInterface") as EventDispatcherInterface;
+
+                            return resolve(this.executeEvent(event, eventDispatcher));
+                        }));
+                    }
+                    break;
+            }
+        })
+
+        // 4- For each event, call the PreResponseMapping Interceptors
+        const eventResponses: EventResponse<any, any>[] = await Promise.all( (await Promise.all(eventsExecutionPromises)).flat().map( async eventResponse => await this.preResponseMappingIntercept(eventResponse)));
+
+        let finalResponse = {};
+
+        // 5 - Construct the final response by calling the events mapper (reverse map method) for each eventResponse;
+        // This method updates the response object that will be returned from the kernel.
+        eventResponses.forEach(eventResponse => {
+            this.eventMappers.forEach( eventMapper => {
+                if(eventMapper.supportsReverseMapping(eventResponse, finalResponse, executionContext)) {
+                    finalResponse = eventMapper.reverseMap(eventResponse, finalResponse, executionContext);
+                }
+            })
+        })
+
+        // 6 - Call the PostResponseMapping interceptors and return the final intercepted response.
+        return this.postResponseMappingIntercept(finalResponse);
```

Comment:
Should we catch errors at a few places if the interceptors throw ?

Reply:
We will let the error bubble up to the Kernel.